src/AvtaleSide/MaalSteg/MaalKort/MaalKort.tsx

```diff
@@ -26,18 +26,22 @@ class MaalKort extends React.Component<Props, State> {
         this.setState({ iEndreModus: modus });
     };
 
-    lagreMaal = (maal: Maal) => {
+    lagreMaal = async (maal: Maal) => {
+        await this.props.lagreMaal(maal);
         this.settEndreModus(false);
-        this.props.lagreMaal(maal);
     };
 
     endreMaal = () => {
         this.settEndreModus(true);
     };
 
+    avbrytEndreMaal = () => {
+        this.settEndreModus(false);
+    };
+
     slettMaal = () => {
-        if (window.confirm("Er du sikker på at du vil slette dette målet?")) {
-            this.props.slettMaal(this.props.maal);
+        if (window.confirm('Er du sikker på at du vil slette dette målet?')) {
```

Comment:
window.confirm('Er du sikker på at du vil slette dette målet?') ? return this.props.slettMaal(this.props.maal) : null;

Reply:
Ikke enig i at ternary er bedre her. Den bør bare brukes når du ønsker å returnere gitt verdi ved `true` og en annen ved `false`, og bør ikke ha sideeffekter. F.eks. `return x > y ? x : y`.

Reply:
Kunne eventuelt ha brukt `&&`, slik: `window.confirm('Er du sikker på at du vil slette dette målet?') && this.props.slettMaal(this.props.maal)`. Men jeg synes ikke det heller er noe bedre enn `if`.